### ISSUE TITLE
Line break is prohibited between CJK if either is enclosed in an environment

### ISSUE THREAD

**Issue by z-rui** [XmacsLabs/mogan#398]
Source:
```
<tabular|<tformat|<cwith|1|1|1|1|cell-width|1in>|<cwith|1|1|1|1|cell-hmode|exact>|<cwith|1|1|1|1|cell-hyphen|t>|<table|<row|<\cell>
  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#800C\>\<#4E0D\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#800C\><with|font-series|bold|\<#4E0D\>\<#4F24\>>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#800C\><line-break><with|font-series|bold|\<#4E0D\>\<#4F24\>>\<#8EAB\>\<#4F53\>\<#3002\>
</cell>>>>>
```

Output:
<img width="116" alt="image" src="https://user-images.githubusercontent.com/7368209/204893940-b78366df-ed58-4275-af1e-736b6a33fbcc.png">

Problem:
- The prohibition of line break between 而 and 不 (which is bold) in the second example is unnecessary.

**Comment by darcy-shen** [comment]
Awesome bug report! This is will solved after #510.

**Comment by darcy-shen** [comment]
A detailed reproducer using Noto CJK and Mogan v1.1.2 beta1 on GNU/Linux:

![图片](https://user-images.githubusercontent.com/1267865/213979244-d4aab923-8a7c-41e1-b9f7-d1a27fb64580.png)

**Comment by darcy-shen** [comment]
This bug is documented in #511 as a subtask of story `[112]`.

**Comment by darcy-shen** [comment]
Unlike #397, this could not be resolved by https://github.com/XmacsLabs/mogan/blob/main/src/System/Language/text_language.cpp

It should not be a subtask of story `[112]`.

**Comment by darcy-shen** [comment]
New Story ID [136] created for this bug.

**Comment by darcy-shen** [comment]
Here is the new bug description:
![图片](https://user-images.githubusercontent.com/1267865/213991177-06c33d4c-dce0-4a84-8197-c8c26473481b.png)

**Comment by darcy-shen** [comment]
after #510 

![图片](https://user-images.githubusercontent.com/1267865/213991342-759f5258-e485-4bb6-b412-ba4710a465d0.png)

It has been solved by #510.
